models/ArticleManager.php

```diff
@@ -0,0 +1,395 @@
+    function post($title, $chapo, $content,$posted){
+        $tableau = [
+            'title'     =>  $title,
+            'chapo'     =>  $chapo,
+            'content'   =>  $content,
+            'writer'    =>  $_SESSION['admin'],
+            'posted'    =>  $posted
+        ];
+
+        $sql = "
+        INSERT INTO articles(title, content, chapo, writer, date, posted, image)
+        VALUES(:title,:content,:chapo,:writer,NOW(),:posted, '')
+        ";
+
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+    }
+
+    // Fonction permettant d'insérer une nouvelle image dans la base de données
+    function post_img($tmp_name, $extension){
+        $id = $this->db->lastInsertId(); //On doit mettre la fonction lastInsertId seulement après une fonction INSERT.
+        
+        $tableau = [
+            'id'    =>  $id,
+            'image' =>  $id.$extension  
+        ];
+
+        $sql = "UPDATE articles SET image = :image WHERE id = :id";
+        
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        move_uploaded_file($tmp_name, "public/img/posts/".$id.$extension);
+    }
+
+        // Fonction permettant de mettre à jour l'image d'un article
+    function update_img($tmp_name, $extension){
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
+         
+        $tableau = [
+            'id'    =>  $id,
+            'image' =>  $id.$extension  
+        ];
+
+        $sql = "UPDATE articles SET image = :image WHERE id = :id";
+        
+        $req = $this->db->prepare($sql);
+        $req->execute($tableau);
+        move_uploaded_file($tmp_name, "public/img/posts/".$id.$extension);
+    }
+    
+    // Fonction permettant de supprimer un article
+    function delete_article()
+    {  
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
+        
+        $query = $this->db->prepare('
+            DELETE
+            FROM articles 
+            WHERE id = :id'
+        );
+        
+        $query->execute(['id' => $id]);
+        
+        unlink("public/img/posts/".$id.".png");
+        unlink("public/img/posts/".$id.".jpg");
+        unlink("public/img/posts/".$id.".jpeg");
+        unlink("public/img/posts/".$id.".gif");
+        unlink("public/img/posts/".$id.".PNG");
+        unlink("public/img/posts/".$id.".JPG");
+        unlink("public/img/posts/".$id.".JPEG");
+        unlink("public/img/posts/".$id.".GIF");
+    }
+    
+      // Fonction permettant de supprimer les commentaires d'un article
+    function delete_article_comments()
+    {  
+        $id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);
+
+        $query = $this->db->prepare('
+            DELETE 
+            FROM comments 
+            WHERE article_id = :id
+        ');
+        
+        $query->execute(['id' => $id]);
+
+        header("Location: /liste-de-tous-les-articles");
+    }
+    function write_verification()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method write_verification() has a Cyclomatic Complexity of 17. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)